File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/C_Code/02.1.1_ButtonLED/ButtonLED.c
/**********************************************************************
* Filename    : ButtonLED.c
* Description : Control led by button.
* Author      : www.freenove.com
* modification: 2019/12/26
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define ledPin    0  	//define the ledPin
#define buttonPin 1		//define the buttonPin

void  main(void)
{
	printf("Program is starting ... \n");
	
	wiringPiSetup();	//Initialize wiringPi.	
	
	pinMode(ledPin, OUTPUT); //Set ledPin to output
	pinMode(buttonPin, INPUT);//Set buttonPin to input

	pullUpDnControl(buttonPin, PUD_UP);  //pull up to HIGH level
	while(1){
		if(digitalRead(buttonPin) == LOW){ //button is pressed 
			digitalWrite(ledPin, HIGH);  //Make GPIO output HIGH level
			printf("Button is pressed, led turned on >>>\n");		//Output information on terminal
		}
		else {							//button is released 
			digitalWrite(ledPin, LOW);  //Make GPIO output LOW level
			printf("Button is released, led turned off <<<\n");		//Output information on terminal
		}
	}
}

File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/Python_GPIOZero_Code/02.1.1_ButtonLED/ButtonLED.py
#!/usr/bin/env python3
########################################################################
# Filename    : ButtonLED.py
# Description : Control led with button.
# Author      : www.freenove.com
# modification: 2023/05/11
########################################################################
from gpiozero import LED, Button

led = LED(17)       # define LED pin according to BCM Numbering
button = Button(18) # define Button pin according to BCM Numbering

def loop():
    while True:
        if button.is_pressed:  # if button is pressed
            led.on()        # turn on led
            print("Button is pressed, led turned on >>>") # print information on terminal 
        else : # if button is relessed
            led.off() # turn off led 
            print("Button is released, led turned off <<<")    

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        print("Ending program")
